Forte.EpiServer.AzureSearch/Events/SearchEventHandler.cs

```diff
@@ -136,7 +137,25 @@ public void OnDeletingContentLanguage(object sender, ContentEventArgs contentEve
             }
         }
         
-        private void UpdateIndexAfterPagePublish(ContentEventArgs contentEventArgs, IContent content)
+        //Event handler for page access rights change
+        public void OnContentSecuritySaved(object sender, ContentSecurityEventArg contentSecurityEventArgs)
+        {
+            var contentLink = contentSecurityEventArgs.ContentLink;
+            var content = _contentLoader.Get<IContent>(contentLink);
+
+            if (content is PageData)
+            {
+                UpdateIndexAfterPageAccessRightsChange(content);
+            }
+        }
+        
+        private void UpdateIndexAfterPageAccessRightsChange(IContent content)
+        {
+            var documentsToIndex = _pageDocumentsProvider.GetDocumentsForAccessRightsChange(content);
```

Comment:
why don't we simply use `_pageDocumentsProvider.GetPageTreeDocuments(content, true)` ?

Reply:
When I created the method, I thought there will be more logic behind it, but GetPageTreeDocuments is not private anyway, so you're right and it's updated now.